### PR TITLE
Use GitHub app instead of GitHub PAT

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details

Instead of using a GitHub PAT as we were before, we will use a GitHub App token, which will GPG sign our commits automatically.

### Related Issues

Related https://github.com/Expensify/Expensify/issues/432173

### CODE REVIEW

File: .github/workflows/npmPublish.yml [hunk 1]
```diff
@@ -11,22 +11,22 @@ on:
         required: true
         type: number
         description: 'Pull request number that was merged to trigger this workflow'
-    secrets:
-      NPM_TOKEN:
```
Comment (AndrewGable) [inline .github/workflows/npmPublish.yml:15]: Using `secrets: inherit` so it will pass this along automatically.

**Review by johnmlee101 — DISMISSED** [review]

**Comment by AndrewGable** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by AndrewGable** [comment]
CLA is going to fail until we merge this unfortunately

**Review by johnmlee101 — APPROVED** [review]

**Comment by AndrewGable** [comment]
Merged and it was going to fail mentioned here: https://github.com/Expensify/GitHub-Actions/pull/2#issuecomment-2405846351

**Pull request merged**